packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts

```diff
@@ -71,22 +71,33 @@ async function fetchCoordinator(callerId: string) {
   return coordinator;
 }
 
-async function validate(coordinator: DbCoordinator, hospital: Hospital) {
+async function validate(
+  coordinator: DbCoordinator,
+  hospital: Hospital | "all"
+): Promise<Hospital[]> {
   switch (coordinator.role) {
     case CoordinatorRole.SYSTEM_USER:
-      return;
     case CoordinatorRole.ZM_COORDINATOR:
     case CoordinatorRole.HOSPITAL_COORDINATOR:
+      if (hospital === "all") {
+        return coordinator.hospitals ?? [];
+      }
       if (!coordinator.hospitals?.includes(hospital)) {
         console.error(
           `Coordinator ${coordinator.id} ${coordinator.role} ${coordinator.hospitals} missing permissions for ${hospital}`
         );
         throw Error("Coordinator has no permissions for hospital");
       }
-      break;
+      return [hospital];
     case CoordinatorRole.GROUP_COORDINATOR:
       break;
   }
+
+  if (hospital === "all") {
```

Comment:
What flow reaches this code?

Reply:
This pattern is prone to errors.

The switch-case should handle all roles, and throw for unfamiliar role. 
It should not return a valid response.

Imagine we decide to add a new role - Guest. That has no permissions at all.
We wouldn't want this role to automatically get permissions to all hospitals, right?

Reply:
got it, working on it..

Reply:
done